library/general/src/lib/yast2/cfa/sysctl.rb

```diff
@@ -0,0 +1,154 @@
+    #
+    # This class does not modify the running kernel configuration. It just writes
+    # the desired values into the configuration file ({PATH}).
+    #
+    # @example Setting a value using they +sysctl.conf+ key
+    #   sysctl = Sysctl.new
+    #   sysctl.load
+    #   sysctl["net.ipv4.ip_forward"] = "1"
+    #   sysctl.save
+    #
+    # @example Setting a value using an accessor
+    #   sysctl = Sysctl.new
+    #   sysctl.forward_ipv4 = "1"
+    #   sysctl.save
+    class Sysctl < ::CFA::BaseModel
+      PARSER = ::CFA::AugeasParser.new("sysctl.lns")
+      PATH = "/etc/50-yast.conf".freeze
+
+      class << self
+        # Defines a key
+        #
+        # Additionally, it adds an accessor that can be used to set the value.
+        #
+        # @param meth [Symbol] Accessor name
+        # @param key  [String] Name of the key used in sysctl configuration files
+        def define_key(meth, key)
+          add_key(key)
+
+          define_method meth do
+            self[key]
+          end
+
+          define_method "#{meth}=" do |value|
+            self[key] = value
+          end
+        end
+
+        # Returns the list of known keys
+        #
+        # Known keys are removed from the old +/etc/sysctl.conf+ when saving the changes.
+        def known_keys
+          @known_keys ||= []
+        end
+
+      private
+
+        # Adds a new key
+        #
+        # @param [String] Name of the key used in sysctl configuration files
+        def add_key(key)
+          known_keys.push(key)
+        end
+      end
+
+      define_key :kernel_sysrq, "kernel.sysrq"
+      define_key :forward_ipv4, "net.ipv4.ip_forward"
+      define_key :forward_ipv6, "net.ipv6.conf.all.forwarding"
+      define_key :tcp_syncookies, "net.ipv4.tcp_syncookies"
+
+      def initialize(file_handler: nil)
+        super(PARSER, PATH, file_handler: file_handler)
+      end
+
+      # Loads sysctl content
+      #
+      # This method reads {PATH} and uses `/etc/sysctl.conf` values as fallback.
+      def load
+        begin
+          super
+        rescue Errno::ENOENT # PATH does not exist yet
+          self.data = @parser.empty
+          @loaded = true
+        end
+
+        self.class.known_keys.each do |key|
+          next if self[key]
+
+          self[key] = Yast::SCR.Read(key_path(key))
+        end
+        nil
+      end
+
+      # Writes sysctl configuration
+      #
+      # Apart from writing the values to {PATH}, it cleans up the same entries from
+      # `/etc/sysctl.conf`.
+      def save
+        super
+        clean_old_values
+      end
+
+      # Returns the value for a given key
+      #
+      # @param key [String] Name of the key to get the value
+      def [](key)
+        data[key]
+      end
+
+      # Sets the value for a given key
+      #
+      # @param key   [String] Name of the key to set the value
+      # @param value [Object] New value
+      def []=(key, value)
+        data[key] = value
+      end
```

Comment:
if this method is needed, I would mark it private to use accessor instead.

Reply:
The only purpose of this module was to open a "backdoor" to set any sysctl value. But after all the keys/attributes handling, now it looks like a bad idea. Removed.